pkg/model/resource/resource.go

```diff
@@ -67,3 +70,20 @@ func (r *Resource) GVK() config.GVK {
 		Kind:    r.Kind,
 	}
 }
+
+func wrapKey(key string) string {
+	return fmt.Sprintf("%%[%s]s", key)
+}
+
+func (r Resource) Replacer() *strings.Replacer {
+	replacements := make([]string, 0, 10)
+
+	replacements = append(replacements, wrapKey("group"), r.Group)
+	replacements = append(replacements, wrapKey("group-package-name"), r.GroupPackageName)
+	replacements = append(replacements, wrapKey("version"), r.Version)
+	replacements = append(replacements, wrapKey("kind"), strings.ToLower(r.Kind))
+	replacements = append(replacements, wrapKey("plural"), strings.ToLower(r.Plural))
+	// To add new replacements just append both the original and the new values and increase the capacity by 2
+
```

Comment:
This comment should not be needed if we change it to unbounded slice (refer to comment above).